pkg/kubeinteraction/cleanups.go

```diff
@@ -45,6 +43,12 @@ func (k Interaction) CleanupPipelines(ctx context.Context, logger *zap.SugaredLo
 			if err != nil {
 				return err
 			}
+
+			// Try to Delete the secret created for git-clone basic-auth, it should have been created with a owneref on the pipelinerun and due being deleted when the pipelinerun is deleted
+			// but in some cases of conflicts and the ownerRef not being set, the secret is not deleted and we need to delete it manually.
+			if secretName, ok := pr.GetAnnotations()[keys.GitAuthSecret]; ok {
+				_ = k.Run.Clients.Kube.CoreV1().Secrets(repo.GetNamespace()).Delete(ctx, secretName, metav1.DeleteOptions{})
```

Comment:
we do have `kubeinteraction` pkg dealing with secret create and update
may be we wanna move this there??!!